Form: Add loading state

jcheringer opened the pull request:
Fixes #28472

## Proposed changes:
* Adds a blur effect to hide the form while it's loading the styles

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Create a post and add a Form block
* Check if the blur effect is being applied correctly when the form is loading the styles. 
* Reloading the page should makes it easier to observe
* Remember to test it in a published post too

nateweller committed:
WAF: Run firewall activation hook on both add_option and update_option (#28234)

nateweller committed:
Release jetpack-waf version 0.8.2 (#28235)

anomiex committed:
cli: Fix ENOENT in `jetpack rsync` (#28200)

If a broken symlink somehow winds up existing in a plugin's `vendor/` or
`jetpack_vendor/`, composer won't clean it up. Then if you try to
`jetpack rsync` the affected plugin, it'll wind up failing with a
cryptic "spawn git ENOENT" when it tries to run git in the nonexistent
target directory.

We can avoid that by checking for broken symlinks before trying to
follow them in the `jetpack rsync` logic. Rsync itself will produce a
more helpful message.

nateweller committed:
WAF: Make Tests Pass (#28185)

renatoagds committed:
VideoPress: Improve "no videos" dashboard to have only one CTA (#28236)

* VideoPress: Update no video dashboard for only one CTA

* changelog

dhasilva committed:
VideoPress: Fix local video listed as VideoPress video (#28237)

* add isLocalVideo flag to VideoRow

* changelog

retrofox committed:
VideoPress: set video by providing a valid GUID value (#28233)

* [not verified] add isVideoPressGuid() helper function

* [not verified] minor TS corection

* [not verified] be able to define a video via GUID

* [not verified] changelog

* [not verified] add a helper function to build URL

* [not verified] handle replace by GUID from replace control

* make attributes optional when building URL

kangzj committed:
Stats: Add loading spinner (#28219)

* [not verified] add loading spinner for stats dashboard

* [not verified] changelog

* [not verified] center spinner

* [not verified] inline style

* [not verified] fix dashboard unit test

* [not verified] bump version

lhkowalski committed:
VideoPress: Route video search using query parameters (#28064)

* Refactor the search params hook to accomodate more features

* Add set, update and reset methods to search params hook

* Use generic setParam to change the pages on the pagination

* Use 1 instead of null to reset the pagination

* Set the search query on the URL when the query changes

* Add changelog file

* Refactor the page handling to use the reset method from search params hook

* Change the query filter when the search term changes on the URL

* Simplify empty string condition from reset function

* Add function to remove parameters from the query string

* Remove empty search filter from URL

* Remove page param when reseting the pagination on the search filter changes

* Remove page parameter when the page is 1

* Remove old query string pages hook

* Set search field value when there is a query string search parameter

* Fix changelog significance

* Remove duplicated setVideosQuery call in favor of the URL-changed one

* Refactor the page validation and page change logic

* Add useEffect hook call to perform the search field value update

* Include the current values of page and search on the useEffect hook dependencies

* Change the approach to set the initial value for the search field

adimoldovan committed:
E2E tests: remove broken selection of complete plan (#28232)

* Remove selection of complete plan in connection flow as it's broken and not used anywhere anyway.

renatoagds committed:
VideoPress: Allow edit privacy on video details page (#28240)

* VideoPress: Add selector for privacy in edit page

* changelog

retrofox committed:
VideoPress: set Video URL in replace control based on privacy (#28239)

* [not verified] set Video URL based on privacy

* changelog

* fix private URL

adimoldovan committed:
E2E tests: fix waitForResponse predicates (#28262)

retrofox committed:
VideoPress: minor TS enhancements in the useSearchParams() hook (#28250)

* set SearchParamNameProp type

* minor jsdoc changes

* changelog

matticbot committed:
Update dependency @storybook/testing-react to v1.3.0 (#28269)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

matticbot committed:
Update dependency qrcode.react to v3 (#28270)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

matticbot committed:
Update dependency react-page-visibility to v7 (#28272)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

coder-karen committed:
Calendar library and Tonesque: various cleanup (#28274)

* [not verified] Calendar library: various cleanup

* [not verified] Correct logic

Co-authored-by: Karen Attfield <karenlattfield@gmail.com>

* [not verified] Update projects/plugins/jetpack/modules/shortcodes/upcoming-events.php

Co-authored-by: Karen Attfield <karenlattfield@gmail.com>

Co-authored-by: Brandon Kraft <public@brandonkraft.com>

retrofox committed:
VideoPress: clean video attributes that are not options when replacing the video file (#28249)

* clean attrs that are not options when replacing

* changelog

* fix setting defauls attribute values issue

* replace block when replacing with a new file

coder-karen committed:
Backporting Jetpack, Social and Backup readmes and changelog entries (#28271)

* [not verified] Updating changelog for 11.7 release

* [not verified] Updating readme.txt for 11.7 release

* [not verified] Updating changelog and readme for 1.4.3

* [not verified] Cleaning up changelog and readme for 1.6.0

* Updating stable tag in readme

* Updating changelog to link to release post

* Updating Social and Backup readme version

* Cleaning up Social and Backup readme and changelog

matticbot committed:
Update storybook monorepo to v6.5.15 (#28268)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

matticbot committed:
Update dependency postcss to v8.4.21 (#28278)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

jeherve committed:
Social Menu / Social Icons Widget: add Mastodon (#28175)

thingalon committed:
[Boost] Don't show <1x on images that are exactly the right size (#28245)

* Mark exact-sized images with a check mark

* changelog

Co-authored-by: Mark George <thingalon@gmail.com>

matticbot committed:
Update dependency storybook-addon-mock to v3 (#28279)

* Update dependency storybook-addon-mock to v3
* Update use of storybook-addon-mock for changes in v3

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Brad Jorsch <brad.jorsch@automattic.com>

thingalon committed:
Remove the Image Guide feature-flag (#28244)

* Enable the Image Guide without a feature-flag

* changelog

Co-authored-by: Mark George <thingalon@gmail.com>

jeherve committed:
Connection: add new method to fetch a connected site's blog ID. (#28208)

danielbachhuber committed:
Form Block: Add a few keywords to the 'Newsletter Sign-up' variation (#28204)

jeherve committed:
Blocks: Add AI-powered blocks for WordPress.com customers (#28043)

Co-authored-by: artpi <artur.piszek@gmail.com>

coder-karen committed:
Revert "Tiled Gallery block: Adding styles to squareish tiled gallery images on WoA sites (#27858)" (#28287)

This reverts commit 5e3d04a31e036cbcb2b2abcd52bafaf1127d7ccf.

retrofox committed:
VideoPress: fix render player once file uploads issue (#28296)

* update timeour reference when cleaning the timer

* changelog

adimoldovan committed:
E2E tests: new test for subscribe block (#28289)

nateweller committed:
Protect: Poll for scan status when scanner is idle and no lastChecked value is available (#28275)

samiff committed:
Jetpack: 11.8-a.1 release (#28298)

* Jetpack: 11.8-a.1 release changelog

* Jetpack: init 11.8-a.2

* Jetpack: refresh prerelease branch before backport

nateweller committed:
Protect: Round ProgressBar percentage values (#28273)

anomiex committed:
build: Use PR commit SHA in plugin artifact version numbers (#28294)

We intend for the version number in the plugin artifacts to include the
short hash for the PR head, but the logic used was picking up the merge
commit GitHub creates instead. Fix that.

samiff committed:
Jetpack: 11.8-a.1 changelog editorial (#28300)

* Jetpack: 11.8-a.1 changelog editorial

* changelog

* Fix missing backtick messing with MD display.

adimoldovan committed:
E2E tests: remove check for post page (#28299)

jeherve committed:
Blaze: only display UI to English customers. (#28266)

renatoagds committed:
VideoPress: Fix app breaking when deleting last page video. (#28281)

* VideoPress: Push back page if deleting last video

* changelog

* VideoPress: Update totalPages and total, and move to action

matticbot committed:
Update react-router monorepo (#28303)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

samiff committed:
Adjust phpcs:ignore usage for wpcom compat (#28304)

dilirity committed:
[Boost] Fix toggling Image Guide from requesting page speed refresh (#28261)

* Fix toggling image guide requesting a page speed score refresh

* changelog

dilirity committed:
[Boost] Update image guide tooltip presentation (#28297)

* Update image guide tooltip documentation link position (bottom) and text

* Add icon to image guide tooltip documentation link

* changelog

* Update description text for small images

* Update changelog

* Update tooltip link color

dilirity committed:
[Boost] Update image guide tooltip link to use jetpack redirect (#28284)

* Fix image guide build not working properly with automattic packages

* Update image guide tooltip url to use jetpack redirect manager

* changelog

* Fix changelog tense

dilirity committed:
[Boost] Mention live chat on premium purchase success page (#28228)

* Update wpcom summary text on purchase successful page

* changelog

* Update description text style to match requirements

retrofox committed:
VideoPress: avoid requesting unneeded preview when block mounts (#28311)

* avoid unneeded preview when block mounts

* changelog

* minor changes in debug() logs

dsas committed:
Make url acceptance more liberal in admin-menus (#28301)

Treat menu items with a fully qualified url matching the site url the
same as other internal URLs in the admin-menus rest api endpoint.

This API endpoint is used for nav-unification on wpcom. This 
change improves compatibility with 3rd party plugins, so that their
menu links work on WordPress.com.

daledupreez committed:
Revert PR #27958 to address conflict with WooCommerce (#28312)

* Revert "Fix out-of-bounds menus on the admin (#27958)"
This reverts commit 2a990efa4b3d7499dbfe8cb461af6064468f8e37.
* changelog

jeherve committed:
Blaze: do not display the UI on private or unlaunched sites. (#28315)

jeherve committed:
Status: add 2 new methods to detect whether a site is private (#28322)

samiff committed:
Jetpack: init 11.8 a.4 (#28327)

* Jetpack: release 11.8-a.3

* Jetpack: init 11.8-a.4

jeherve committed:
Blaze: do not load the UI if the JSON API module is inactive (#28267)

dhasilva committed:
VideoPress: Enhance behavior when deleting multiple videos (#28302)

* refactor delete action adding FLUSH_DELETED_VIDEOS

* changelog

* fix removedVideoCount processing order

* enhancements from code review

* remove videosBeingRemovedCount

retrofox committed:
VideoPress: fix setting title when uploading video file (#28329)

* [not verified] move title handling to uploader progress cmp

* [not verified] minor tidy changes

* [not verified] remove local state to set the title

* [not verified] order the updating requests

* [not verified] try to update the poster asap

* changelog

romarioraffington committed:
Add User Stories for earn features (#28283)

* Add regression testing instructions for post level visibility setting
* Add regression testing instructions for premium content

wilmersondasilva committed:
Odyssey Stats: Fix WP Footer Position (#28308)

* [not verified] Odyssey Stats: remove inline CSS that limit the height of  #wpcom div

This was making the Odyssey Stats having some UI issues, like the footer that was wrongly positioned

* [not verified] changelog

* [not verified] Odyssey Stats: fix inline CSS that define the height of the #wpcom div

This was making the Odyssey Stats having some UI issues, like the footer that was wrongly positioned

* [not verified] Odyssey Stats: fix changelong significance

* Odyssey Stats: update version on package.json and class-main.php

candy02058912 committed:
Blaze: Laying foundations for showing Advertising in menu for Jetpack sites (#28088)

* Blaze: Show Advertising in menu for Jetpack sites

* changelog

* Add unit test

* Fix unit test

* Add locale restriction for blaze

* Use user locale

* Update unit test and locale restrictions

* Fix the connection manager code

* Use blaze package

Co-authored-by: sdixon194 <steve.dixon@automattic.com>

thingalon committed:
[Boost] Clean up leftover image guide gating. (#28306)

* Remove image-guide gates from a few spots I missed

* changelog

Co-authored-by: Mark George <thingalon@gmail.com>

retrofox committed:
VideoPress: tweak the footer of the uploader component (#28337)

* debounce generating frame call

* tweak uploader component footer

* changelog

retrofox committed:
Jetpack: fix VideoPress fullscreen option when using shortcode (#28342)

* fix setting fullscreen player when using shortcode

* changelog

mashikag committed:
JSON Endpoint: Create `GET /sites/%s/dropdown-pages/` endpoint (#28132)

* JSON Endpoint: Create 'GET /sites/%s/dropdown-pages' endpoint

*  changelog

* Fix return for the endpoint

* For each page return 'title' instead of 'post_title' prop.

* Instead of making direct db query make a call to get_pages. Return a list of tree-like page objects.

* Ignore unused variable warning

* Correct condition to check if page has a parent

* Reverse the if clause for building pages_by_parent map

* Fix up types

* Update projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-dropdown-pages-endpoint.php

Co-authored-by: Ivan Ottinger <25105483+ivan-ottinger@users.noreply.github.com>

* Update projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-dropdown-pages-endpoint.php

Co-authored-by: Ivan Ottinger <25105483+ivan-ottinger@users.noreply.github.com>

* Update projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-dropdown-pages-endpoint.php

Co-authored-by: Ivan Ottinger <25105483+ivan-ottinger@users.noreply.github.com>

* Update projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-dropdown-pages-endpoint.php

Co-authored-by: Ivan Ottinger <25105483+ivan-ottinger@users.noreply.github.com>

Co-authored-by: Ivan Ottinger <25105483+ivan-ottinger@users.noreply.github.com>

mashikag committed:
Site Settings API: Expose `show_context` and `show_date` options (#28305)

* [not verified] Site Settings API: Expose related posts

* [not verified] Site Settings API: Expose related posts  option

* [not verified] changelog

* [not verified] Fix return value on related posts option update

* [not verified] Handle undefined  or  options

* [not verified] Remove unnecassary bool cast

* [not verified] Update projects/plugins/jetpack/changelog/update-expose-related-posts-show-date-option

Co-authored-by: Ivan Ottinger <25105483+ivan-ottinger@users.noreply.github.com>

* [not verified] Bring back the bool cast

* [not verified] Fix up some options assignments

Co-authored-by: Ivan Ottinger <25105483+ivan-ottinger@users.noreply.github.com>

adimoldovan committed:
E2E tests: disable blocks with Gutenberg suite (#28343)

coder-karen committed:
Add CRM to the Monorepo (#28314)

Co-authored-by: gogdzl <diego.rodrigues@automattic.com>
Co-authored-by: Brad Jorsch <brad.jorsch@automattic.com>
Co-authored-by: sdixon194 <steve.dixon@automattic.com>

artpi committed:
Jetpack AI: Generate a post intro based on title and categories (#28321)

This PR makes the AI Paragraph block include the contextual information for a better prompt if content for completion is not available

artpi committed:
[Jetpack AI] Easier enabling for testing, forums link (#28338)

* Centralize the switch to enable on/off

* Change icons and link the forum thread

* changelog

* Handle the disappearing block

* Fix lint warnings

* Filter comment

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

* Add a Jetpack redirect

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

retrofox committed:
VideoPress: tweak video title and description (#28341)

* [not verified] add decodeEntities() helpoer function

* [not verified] decode title and description once data got

* [not verified] changelog

* [not verified] add MDN reference to createTextareaelement() jsdoc

* rewird changelog message

* return always the input string

* add reference ad see tag

* Update projects/packages/videopress/src/client/lib/url/index.ts

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

sdixon194 committed:
Allow Avengers to Approve CRM Code (#28348)

dhasilva committed:
VideoPress: Add rating selector on video details page (#28347)

* add rating selector on video details edit page

* changelog

anomiex committed:
tools: Fix check-plugin-monorepo-deps.sh with no monorepo deps (#28324)

If there are no monorepo package deps, the script prints out an error
like

  crm:  should be at dev-trunk, not

That shouldn't happen, add a check for the situation to avoid it.

pyronaur committed:
Jetpack Boost: Image Guide compatibility improvements (#28026)

* Use a fixed position for the guide

* Increase padding

* Add a Portal component that teleports slot to body

* Portal the pop-up to body to fix z-index issues

* FIxed a compatibility bug with Jetpack Mosaic gallery

Wrapping caused the object-fit to break, this seems to be a more lightweight approach.

* A hacky compatibility fix across plugins

* Simplify keeping the Pop-up open using a class

* Rename classes

* Don't overlay other interactive elements as much as possible

By taking up as little space as possible.

* Be a little more clear on what's going on, WIP

* changelog

* Undo inexplicable change from merge

* Update info block position in tooltip

* Fix image guide inside links

* Refactor getClosestContainingAncestor to not use arrays to keep track of current element

* Refactor getClosestContainingAncestor to improve code readability

* Disable image guide in iframes

This is done to ensure editors using iframes don't initialize
the image guide in the WP-Admin.

* Lints

* Revert regex update that happened during trunk merge

Co-authored-by: Mark George <thingalon@gmail.com>
Co-authored-by: Peter Petrov <peter.petrov89@gmail.com>

matticbot committed:
Update dependency gridicons to v3.4.1 (#28360)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

matticbot committed:
Update dependency social-logos to v2.5.1 (#28361)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

thingalon committed:
Extra options to configure cache htconfig (#28031)

* Allow overriding htaccess files using filters

* changelog

* Remove syntax deprecated in PHP 8.2

* Quote header values at output time

* Remove automatic mapping of ExpiresByType setting based on Cache-control setting

Co-authored-by: Mark George <thingalon@gmail.com>

sergeymitr committed:
Connection: add support for the 'connection_disabled' error code. (#28334)

ouikhuan committed:
Introduce Jetpack Migration plugin (#28209)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

anomiex committed:
composer-plugin: Ensure `jetpack_vendor/` exists before creating i18n-map (#28369)

If something uses the plugin but does not use any "jetpack-library"
packages, `jetpack_vendor/` won't exist and the plugin will fail when
trying to create the `jetpack_vendor/i18n-map.php` file. Account for
this situation by creating the directory if necessary.

retrofox committed:
VideoPress: add anchor support to VideoPress video block (#28377)

* add anchor support to v6

* changelog

coder-karen committed:
Releasing new versions of actions/packages and the Beta plugin (#28290)

danielbachhuber committed:
Docs: Include mention of `jetpack_blocks_variation` filter (#28382)

dereksmart committed:
Required Review: Allow anyone who can approve to also approve docs/ (#28359)

Co-authored-by: Derek Smart <dsmart@dsmart-mbp.lan>

jeherve committed:
Blaze: rely on method instead of filter to check for eligibility (#28340)

millerf committed:
Add/openai placeholder prompts (#28339)

* Add placeholder and use it to prompt if empty
* changelog
* Updated for Dall-E prompts
* Push examples to a separate file
* change placeholder for each new block entity
* Added ChatGPT prompts

retrofox committed:
VideoPress: skip rating checking when pulling video data for block (#28374)

* [not verified] add skipRatingChecking option to useVideoData hook

* [not verified] skip rating checking when pulling video data

* changelog

* rename option to skipRatingControl

retrofox committed:
VideoPress: use core lib to handle html entities (#28376)

* add @wordpress/html-entities dep

* decode title and description html entities

* dont invent the weel again. Renove decodeEntities

* changelog

* Update projects/packages/videopress/changelog/update-videopress-use-core-pkg-to-handle-entities

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

kallehauge committed:
CRM: Introduce Jetpack Autoloader and declare Dompdf in Composer (#28364)

samiff committed:
Jetpack: backport 11.7.1 changelog (#28393)

* Jetpack: backport 11.7.1 changelog

* changelog

samiff committed:
Jetpack: backport 11.8-a.5 changelog (#28395)

* Jetpack: 11.8-a.5 changelog

* Jetpack: init 11.8-a.6

samiff committed:
Jetpack: 11.8-a.5 changelog editorial (#28396)

* Jetpack: 11.8-a.5 changelog editorial

* changelog

thingalon committed:
[Boost] Fix woocommerce deprecation warnings (#28373)

* Clean up woocommerce deprecation warnings

* changelog

Co-authored-by: Mark George <thingalon@gmail.com>

pyronaur committed:
Image Guide Cleanup (#28381)

* Fix variable capitalization

* Update grammar

* Always on state isn't compatible with portals

If we want to provide this functionality in the future, we might need to create a more comprehensive "Page report" instead.

* jetpack-boost-image-guide -> jetpack-boost-guide

* Set the initial "show" state to false

* Changelogger entry

jeherve committed:
Blocks: update recipe category block (#28378)

artpi committed:
[Jetpack AI] More transparent prompt engineering for content completion (#28351)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

pyronaur committed:
Bump Jetpack Boost version to 1.7.0 (#28404)

* Optimise -> Optimize

* Add a changelogger entry

* Bump the versions

creativecoder committed:
Writing prompts: remove the placeholder prompt and setting (#28387)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

mashikag committed:
Settings API: Allow to un-assign `page_for_posts` and `page_on_front` on `POST` request (#28379)

* Settings API: allow for unassigning 'page_on_front' and 'page_for_posts' options

* changelog

* Make response value null for the deleted option

* Enhance code explanation comment

* Update projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

Co-authored-by: Ivan Ottinger <25105483+ivan-ottinger@users.noreply.github.com>

Co-authored-by: Ivan Ottinger <25105483+ivan-ottinger@users.noreply.github.com>

dilirity committed:
Prepare for 1.7.0 (#28407)

ivan-ottinger committed:
Add `default_option_subscription_options` filter (#28320)

The purpose of this filter is to introduce the default value for `subscription_options`.

This default value will be available throughout Simple, Atomic and Jetpack sites, including the Site Settings endpoint.

retrofox committed:
VideoPress: check source language length of the video chapter (#28406)

* tidy import types section

* show notice error when lang is too long

* tweak notice error

* do not allow upload track when error

* changelog

* update version

* udpate pnpm deps

BogdanUngureanu committed:
Google Fonts Provider: Fix Global Styles font collection (#28411)

* [not verified] Use wp_get_global_styles() and fallback to gutenberg_get_global_styles

* [not verified] Removed autoformatting changes

* Bump version so that changelogger checks won't fail.

ice9js committed:
Add the skeleton for jetpack/forms (#28409)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

adimoldovan committed:
E2E tests: handle block editor frame when using Gutenberg (#28386)

romarioraffington committed:
Premium Content / Subscriptions: introduce Paid Newsletter functionality ( 2nd Attempt ) (#28170)

Introduce paid-newsletter functionality gated by `JETPACK_BETA_BLOCKS`

samiff committed:
google-fonts-provider: release v0.4.1 (#28418)

samiff committed:
Jetpack: init 11.8-a.8 (#28420)

* Jetpack: release 11.8-a.7

* Jetpack: backport 11.8-a.7 and init 11.8-a.8

anomiex committed:
crm: Set up infrastructure to run tests in CI (#28354)

Import the tests from the old private repo, then set up the
infrastructure to run them within our CI.

Also fix issues with the tests necessary to make them pass.

Co-authored-by: gogdzl <37049295+gogdzl@users.noreply.github.com>
Co-authored-by: cleacos <javacodia@gmail.com>

simonktnga8c committed:
Add Protect paid feature upgrade info (#27925)

Now that Protect has malware scanning and removal as a paid feature, including this text in the readme for the plugin listing.

CGastrell committed:
Jetpack: Change responses export filename (#28413)

* add helper function to get export filename

* use filename from response headers (crafted on backend)

* changelog

gmjuhasz committed:
Add Advanced Plan to Social pricing table (#28258)

* Add advanced plan to pricing table + update design

* changelog

* Remove cycle from unlimited shares text

* Add default value to new prop

* Change term to Recycle content based on design

* Fix minification issue with production build

* Fix translation issues

* Add tooltips for features

* Add coming soon type to PricingTableItem

* Fix type

* Change order

Co-authored-by: Siddarthan Sarumathi Pandian <siddarthan@outlook.com>

retrofox committed:
VideoPress: set video block availability, based on site plan (#28391)

* [not verified] add videopress/video support

* [not verified] do not set videopress/video-chapters availability

* [not verified] set videopress/video availability

* [not verified] changelog

* [not verified] rely on availabilty based on site plan

anomiex committed:
Jetpack: Replace deprecated WP function use (#28434)

WordPress 6.2 deprecated `WP_Theme_JSON_Resolver::theme_has_support()`
in favor of `wp_theme_has_theme_json()`. Use it when available.

jeherve committed:
Blocks: move away from deprecated methods in multiple blocks (#28415)

sergeymitr committed:
Debug Helper: add cookie state faker (#28371)

* Debug Helper: add cookie state faker.

* Styling and linter fixes.

jeherve committed:
Sharing: do not include buttons in REST API responses (#28412)

anomiex committed:
tools: Avoid double-tagging in wporg SVN (#28437)

The command used to tag a release in SVN has an unfortunate failure
modde if the tag already exists: instead of simply failing, it will
create a `trunk` directory inside the tag.

This PR does two things to try to avoid this situation:

* The manual `deploy-to-svn.sh` script will now confirm if someone tries
  to run it on a plugin that is set up for auto-publish.
* Both that and the auto-publish will abort if they detect the tag
  already exists (but note this check is subject to races).

millerf committed:
[AI image] fix styling  issues (#28380)

* [not verified] Fix textarea's width

* [not verified] Add changelog

* [not verified] Small refactor

* [not verified] Wrap results

* [not verified] Small refactor. Saves image in a separate function and clean up some CSS

* [not verified] DOM cleanup

* Fix small bug

* Bigger textarea

* Better Flex handling. Add modal

* Fix build and text

* Add button to modal

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

jeherve committed:
General: start using new Status utilities to detect site status (#28328)

* General: start using new Status utilities to detect site status

These methods were introduced in #28322. Let's switch to them.

* Remove unneeded PHPCS exception

* Status: add new filters for the new methods added in #28322

These filters will be helpful for manipulating tests, for example.

* Update test to use new filters

jeherve committed:
Widget Visibility: add missing jetpack Config external (#28402)

matticbot committed:
Update dependency automattic/wordbless to v0.4.1 (#28440)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

retrofox committed:
VideoPress: fix replace video from toolbar issues (#28432)

* propagate new vide data when replacing video

* changelog

* fix issue when picking guid from media item

bindlegirl committed:
Carousel: check array to avoid PHP warnings (#28388)

ice9js committed:
Hook up packages/jetpack-forms to plugins/jetpack (#28416)

bindlegirl committed:
SAL: remove error logging (#28442)

matticbot committed:
Update Eslint packages (major) (#28441)

* Update Eslint packages
* Override outdated eslint-plugin-wpcalypso dep
* Fix new sniff failures

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Brad Jorsch <brad.jorsch@automattic.com>

CGastrell committed:
Change responses export field names (#28425)

* [not verified] prevent last csv line to be a single zero (0)

* [not verified] use Date and Consent as meta field keys on export

* [not verified] use 'IP Address on mapped fields resolution

* [not verified] remove numeral prefix on field/column names before exporting

* [not verified] changelog

* [not verified] remove first column 'Contact Form'

* [not verified] update tests for contact-form

* update moar tests

* use _x for noun specific translation

* More clear changelog entry and description

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

Co-authored-by: Brandon Kraft <public@brandonkraft.com>
Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

dhasilva committed:
VideoPress module: Add test for generated embed code (#28358)

* add test for videopress generated embed html

* changelog

Co-authored-by: Damián Suárez <rdsuarez@gmail.com>

Commit:
Prerelease (#28443)

* Prepare release 1.3.0 of Jetpack VideoPress standalone

* Merge trunk and fix versions

* correct versions

* move versions back to 1.2.1

dhasilva committed:
Revert "VideoPress: fix replace video from toolbar issues (#28432)" (#28447)

This reverts commit 9439f40a7e5f10b57f14a6d4dae15b8921f55f4b.

sergeymitr committed:
Connection: remove handling of the 'site_blacklisted' error code. (#28436)

Remove the error code `site_blacklisted` from the handler, as it's no longer being returned by WPCOM.

matticbot committed:
Update dependency @vercel/ncc to v0.36.0 (#28452)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

samiff committed:
Remove plugins/jetpack-mu-wpcom from the monorepo (#28445)

* Remove plugins/jetpack-mu-wpcom from the monorepo

Will create jetpack-mu-wpcom as a package instead.

* Update pnpm-lock

samiff committed:
Add jetpack-mu-wpcom package skeleton (#28449)

* Add initial jetpack-mu-wpcom package skeleton

* Update gitignore and composer.json for WP tests

* Add changelog

Commit:
Prerelease for VideoPress standalone 1.3 (#28453)

* Prepare release 1.3.0 of Jetpack VideoPress standalone

* Add prerelease changelog

anomiex committed:
tools: Handle v-prefixed versions in stable-tag.sh (#28455)

If the GH version tag has the "v" prefix, that should be stripped before
comparing it with the SVN version and readme versions.

adimoldovan committed:
E2E tests: fix tiled gallery block test with GB 15.0.0 (#28454)

trakos committed:
Fix exceeding search limits in Jetpack Complete (#28115)

* Search: fix exceeding search limits in Jetpack Complete

A site that has Jetpack Complete with exceeded Jetpack Search limits cannot use the instant search feature.
We should not allow toggling the instant search option,
the same way we don't allow it for Jetpack Search Free that is over limit.

In the past, must_upgrade didn't check for the plan that blog has,
but now it's safe to base the decision to disable instant search option only on that parameter.

This commit introduces using `must_upgrade` to determine whether to disable instant search toggle.

I've also added one final CTA that's shown to both Jetpack Search Free and Jetpack Complete users,
shown once their search plan is disabled after three months of overages.

Additionally, we will automatically disable instant search and search module on visiting Jetpack admin
when `must_upgrade` is true (exceeded search limits).
This shouldn't be needed most of the time, but occasionally the API call to disable it from overage cron might fail.

Commit:
Image Guide: Bundle image guide (#28429)

* Bundle image-guide into a single file

* Make the bundle usable in boost

* changelog

* Update changelogs

* Add a library name

* Update library variable name

* Update project versions

Commit:
Create changelog (#28456)

retrofox committed:
VideoPress: fix replace VideoPress video  (#28451)

* propagate new vide data when replacing video

* changelog

* fix issue when picking guid from media item

* fix setting attrs when uploading file

* bump version

ivan-ottinger committed:
Fix Related Posts options saving in the endpoint (#28435)

arthur791004 committed:
Add a new endpoint template-loader to resolve the correct template by type (#28158)

anomiex committed:
build: Move "Test plugin upgrades" to a post-build workflow (#28241)

GitHub has an unfortunate behavior in that workflow artifacts are not
available via the API until the entire workflow has completed. This
means that the "Test plugin upgrades" job in the build workflow
unnecessarily delays the availability of the build artifact for Jetpack
Live Branches and future wpcom automated testing for 6 or so minutes.

To get around this, we create a new "Post-Build" workflow that is
triggered by `workflow_run` after the Build workflow completes. But
since this new workflow isn't automatically attached to the PR, we have
to make API calls to get it to show up there.

This is a second attempt at #25892, which had to be reverted as using
the Actions app put the checks into the "suite" of a random workflow.

jeherve committed:
Blaze: always enqueue Connection state info (#28457)

anomiex committed:
build: Restore "Test plugin upgrades" in build job (#28467)

Something is not working right yet with the post-build job. Let's keep
it around for debugging, but restore the copy in the build job so
peoples' PRs aren't blocked by it.

samiff committed:
Release packages/jetpack-mu-wpcom v0.1.0 (#28469)

jcheringer committed:
Add blur effect while the Form is loading styles

jcheringer committed:
Add changelog

retrofox committed:
VideoPress: improve buildVideoPressURL(). Add tests. (#28465)

* [not verified] handle files.URLs

* [not verified] add tests for URL lib

* [not verified] buildVideoPressURL() return empty object

* change buildVideoPressURL() return value

* changelog

* escape . in the RegExp

* escape the second dot :facepalm:

retrofox committed:
VideoPress: do not prompt to convert embed block to VideoPress video block (#28474)

* [not verified] remove the filter that prompts to conver embed

* [not verified] changelog

* [not verified] re-organize transforms code

* [not verified] minor tidy

* [not verified] clean up classes when transforming from Embed

* [not verified] set explicitely attrs when transform to embed

* [not verified] tweak to Embed transform

bindlegirl committed:
Infinite scroll: AMP related bug fix (#28466)

retrofox committed:
Jetpack: do not add block transform to v6 when video is not a VideoPress video (#28477)

* [not verified] improve fn to pick the GUID from URL

* [not verified] do not transform when is not a videopress video

* changelog

sdixon194 committed:
Releases: Add a script that does a full release (#28048)

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

kangzj committed:
Stats: enable ads page for Odyssey (#27791)

* [not verified] try: enable Ads tab for Stats

* [not verified] changelog

* [not verified] Show Ads tab for Odyssey

* [not verified] add wordads local controllers

* [not verified] tweak wordads controllers

* [not verified] changed supported version to 0.4.0-alpha

* [not verified] fix up versions

* [not verified] ignore WordPress.Security.NonceVerification.Recommended

* [not verified] remove todo comment

* [not verified] ignore unknow capatibility

* [not verified] fix typo

* [not verified] remove  unecessary version test

nateweller committed:
Protect: Fix API response property name case styling (#28397)

vishnugopal committed:
Add viewers to /users endpoint when include_viewers is true (#28317)

PT: pcmemI-1uJ-p2
Specific finding that led to this diff: pcmemI-1wa-p2#comment-1276

For simple sites, our new team page does not display viewers currently. This is because for simple sites, viewers are an entirely different implementation in a different table, and is normally accessed through a corresponding /sites/$site_id/viewers endpoint.

Since it's clumsy on the frontend to call these two endpoints and combine data, we're introducing a new include_viewers argument to /sites/$site_id/users/ which when set to true:

- only supports searching based on name for viewer
- ignores order and order by parameters for the entire list and sorts by name

This implementation should satisfy our current requirement while being entirely backwards compatible.

retrofox committed:
VideoPress: tweak uploader layout of the VideoPress video block (#28482)

* tweak uploader component layout

* changelog

Commit:
Jetpack: Include Boost on "My Products" (#28336)

* Add boost in My Plans

* changelog

* Fix yearly boost plan detection

retrofox committed:
VideoPress: remove undesired border of video player (#28484)

* clean up the border of the video player

* changelog

renatoagds committed:
Components: Fix responsive across elements with box-sizing removal (#28489)

* My Jetpack: Remove local box-sizing

* Components: Add box-sizing at ThemeProvider

* changelog

* Bump version

* Components: Rename class from theme to global

digitalwaveride reviewed (approved):
Thanks João! 👍 
Works for me, and fixes our issue for now when styles loading is slightly delayed.

adimoldovan committed:
E2E tests: wait for editor canvas load (#28494)

retrofox committed:
VideoPress: do not request video data when user is not connected (#28493)

* [not verified] expose connection initial state in the client side

* [not verified] do not perform sync request when no connected

* changelog

pyronaur committed:
Initialize the WP JS Data Sync package (#28495)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

jeherve committed:
Reviewer list: add Heart Of Gold for new packages (#28497)

anomiex committed:
actions: Add some error detection in the check-run action (#28470)

The existing code just proceeds with an ID of "null", which was not
intended.

thingalon committed:
[Super Cache] Add missing action trigger (#28481)

* Add a missing call to wp_cache_Cleared

* changelog

Co-authored-by: Mark George <thingalon@gmail.com>

pyronaur committed:
Svelte JS Data Sync Client (#28496)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

artpi committed:
[Jetpack AI] Fix regression when loading the saved block editor  (#28488)

* Fix a regression when loading the editor
* changelog

anomiex committed:
ci: Remove slack notice for post-build workflow (#28502)

It's complaining on every merged PR and we can't fix it until GitHub
gets back to us to say what is going wrong.

anomiex committed:
actions: Replace getsentry/action-github-app-token with custom code (#28507)

It seems likely we're running into getsentry/action-github-app-token#18.

Let's replace it with some custom code that works correctly locally.

bindlegirl committed:
CDN: deprecated error fix (#28508)

coder-karen reviewed (approved):
Works brilliantly! 
Apologies for the 'force push' mess (now removed), had some issues whilst trying to rebase for some reason.

coder-karen committed:
Tooling: Wording addition to deploy-to-svn script (#28506)

jeherve committed:
Repo experience: update report template & automated triage task (#28448)

The GitHub issue template is updated in Calypso in https://github.com/Automattic/wp-calypso/pull/71827/

Let's update ours to match and provide a consistent experience. We should also update our GitHub action that looks for key content in GitHub issues, so it knows where to find info about impacted platform and priority indicators in the updated bug report format.

fgiannar committed:
Jetpack Sync: prevent syncing user update when corresponding flags are not set. (#28385)

* Jetpack Sync: Prevent syncing user update when flags are not set

bisko committed:
Fix/sync close php session immediately (#28515)

* Close active PHP Session if found when running a Dedicated Sync request, to make sure we're not locking the request unnecessarily.

* Add changelog

anomiex committed:
cli: Update WorDBless install commands (#28517)

Looks like we never updated the CLI generate command with the updated
command WorDBless uses, so new projects were getting generated with the
wrong code. Fix that, and the affected projects.

Also fill in the necessary `config.allow-plugins` in the generated
composer.json.

jkguidaven committed:
JITM: Prevent the activation page from displaying the JP License Activation JITM. (#27959)

* [not verified] Fix JP License Activation JITM to not display in Activation page.

* [not verified] changelog

* [not verified] Remove logic for filtering notice within jitm package and instead push message path with hash for all My Jetpack page.

* [not verified] Fixup changelog description and package version.

jeherve committed:
Blaze: add new method to remotely check for eligibility (#28353)

* Blaze: add new method to remotely check for eligibility

This will allow us to remotely check eligibility for a site, and potentially update rules without having to release a new version of the package.

* Remove logging function

* Remove to-do comment

* fix tests by bypassing remote request in tests.

* Fix return value for tests

* Make new method public

See https://github.com/Automattic/jetpack/pull/28353#discussion_r1071281414

* Bypass API request on WordPress.com

There, we can get the response directly.

* Remove private site check

This can now be done on WordPress.com.

* Simplify list of checks by wrapping Jetpack-specific checks

This will limit the number of checks we have to run on WordPress.com, and will also help with the tests.

jeherve committed:
Block bundling: introduce single variation constant (#28390)

* Block bundling: introduce single variation constant

Fixes #28227

Instead of having 2 separate constants to manage what block bundle is loaded on a site, let's only use one, JETPACK_BLOCKS_VARIATION. It makes it easier to manage and overwrite.

I also sunset 2 existing filters in favor of the existing jetpack_blocks_variation filter which works the same way as our new constant.

* Continue to support old constant

* Update changelogs' wording

Co-authored-by: Daniel Bachhuber <daniel.bachhuber@automattic.com>

* ver bump

Co-authored-by: Daniel Bachhuber <daniel.bachhuber@automattic.com>
Co-authored-by: Karen Attfield <karenlattfield@gmail.com>
Co-authored-by: Brandon Kraft <public@brandonkraft.com>

jeherve committed:
Blaze: add new Tracks even to track panel views. (#28392)

* Blaze: add new Tracks even to track panel views.

It will allow us to build more accurate funnels for that flow.
See p1673443048626029-slack-C04DZ725FEK

* Consolidate checks to display panel and track view

1. Remove some redundant checks now that the panel's js doesn't even get enqueued in some scenarios (we've added conditions to load the js directly in PHP)
2. Add checks to see if a post was just published, not only if it is already published.
3. Use said checks to control display of the panel as well as track view.

* Add some comments

* Bump version

Co-authored-by: Brandon Kraft <public@brandonkraft.com>

dereksmart committed:
Update E2E test account password (#28519)

Co-authored-by: Derek Smart <dsmart@dsmart-mbp.local>

IanRamosC committed:
Fix Intro Offer price description (#28424)

* Fix how to calc Jetpack Backup intro offer price

* Add intro offer object to ProductDetailCard

* changelog

* Backup: fix PricingCard currency and details text

* Products: only return intro offer when available

This commit fixes the return of Wpcom_Products::get_product_pricing to only return `introductory_offer` when available

* Products: fix test_get_product_price

* Fix version mismatch

* Backup: revert intro offer calculation

* My Jetpack: revert monthly price calculation

* My Jetpack: fix price detail text condition

* Backup: fix price detail text condition

* Fix changelogger version

* Fix bad minification issue on production builds

kraftbj committed:
Merge remote-tracking branch 'origin/trunk' into update/form-loading
